tests/rifCollectorAppTest.php

```diff
@@ -42,4 +44,38 @@ public function testPopulateTableMalformed(): void
         $this->expectException(TypeError::class);
         populateTable($input);
     }
+
+    public function testValidateSanitizeEntrySuccess(): void {
+        $inputtedArray =
+            ['make' => 'MAKE',
+            'model' => 'MODEL',
+            'type' => 'TYPE',
+            'color' => 'tan',
+            'mags' => 2,
+            'power' => 'POWER SOURCE',
+            'sites' => 1,
+            'purchased' => '2024-01-01'];
+
+        $expected =
+            $validatedSanitizedArr =
+                ['make' => 'MAKE',
+                'model' => 'MODEL',
+                'type' => 'TYPE',
+                'color' => 1,
+                'mags' => '2',
+                'power' => 'POWER SOURCE',
+                'sites' => '1',
+                'purchased' => '2024-01-01'];
+
+        $actual = validateSanitizeEntry($inputtedArray, $validatedSanitizedArr, $message);
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    public function testValidateSanitizeEntryMalformed(): void {
+        $input = 'oh no I am going to break it';
+
+        $this->expectException(TypeError::class);
+        populateTable($input);
+    }
```

Comment:
This is more of a failure test - you give it not even the data type and expect a failure. It's a 'not even wrong' test. A more informative malformed test would provide inputs that you think *should* fail validation - a color that isn't a numerical ID for example, or a date that doesn't pass your data format regex.